### DISCUSSION TITLE
Yhdys-merkin hyväksyminen kaiken-laisissa yhdys-sanoissa

### DISCUSSION THREAD

**Discussion by yucca42** [voikko/discussion#11] [General]
Käsitykseni mukaan suomen kielen normit sallivat (ja ovat vaihtelevantyylisesti aina sallineet) yhdysmerkin käytön yhdyssanan osien välissä selvyyden vuoksi, esimerkiksi ”sisar-ilta”. Ohjeita kirjoitettaessa on ajateltu suhteellisen ahtaasti sellaisia ta­pauk­sia, joissa olisi muutoin vaikea hahmottaa, missä yhdysosien raja on, tai ylipäänsä huo­ma­ta sana yhdys­sanak­si. Sääntöjen sana­muodot antavat kuitenkin oikeuden käyttää yhdys­merk­kiä laajasti.
https://jkorpela.fi/kielenopas/4.4.html#yhd-selv

Katsoin, miten Oikofix käsittelee tuolla sivullani siteeratun listan tapauksista (Kielitoimiston oikeinkirjoitusoppaasta).  Muun muassa ”sisar-ilta” menee läpi ja analysoituu yhdyssanaksi, mutta rap-artisti, syys-aiheinen, nyky-isyys ja päivä-isyys eivät. Ilmeisesti ne eivät lainkaan tunnistu sanoiksi. Rap-artisti on sikäli selvä tapaus, että rap-sanalla on lippu ”ei sallittu yhdyssanan osana” (syynä kai se, että muuten rap-alkuiset sanat tunnistuisivat liian herkästi yhdyssanoiksi), eikä lippua ”ei sallittu yhdyssanan osana, paitsi yhdysmerkillä erotettuna” taida vielä olla olemassa. Muiden osalta tunnistumattomuuden aiheuttaa juuri yhdysmerkki; ilman sitä ne tunnistuvat oikein.

Olisiko mahdollista sallia aina yhdysmerkki yhdyssanan osien välissä, ja mitä se vaikuttaisi? Oikoluvussa ei enää saisi virheilmoituksia sellaisista asuista kuin ”päivä-kahvi”, jossa yhdysmerkki on varmaan useimpien mielestä tarpeeton, monien mielestä väärä – ja silti kielen normien mukainen.

**Comment by ahomansikka** [comment]
Tiedostojen indeksoinnissa Sukija hyväksyy kaikki yhdys-sanat (ja sivuvaikutuksena myös ta-vu-tuk-set). Ylimääräisten viivojen käsittely on tehty Javalla, enkä tiedä, miten iso remontti on laittaa se Voikon foma-koodiin.

**Comment by ahomansikka** [comment]
Palatakseni asiaan. Solr:n Javalla kirjoitetussa Sukija-liitännäisessä asia on hoidettu niin, että jos Voikko ei tunnista merkkijonoa ja siinä on yhdysviiva, se otetaan pois ja yritetään uudelleen. Oikoluvussa sama asia voitaneen tehdä C/C++:lla puukottamatta foma-koodia.

**Comment by tlikonen** [comment]
Yhdysmerkin salliminen aina kaikissa yhdyssanoissa tuntuu minusta liian paljolta, liian hyväksyvältä, koska nähdäkseni suurin osa ihmisistä pitää niitä väärinä. Sellainen olisi kyllä ihan mainio valinta sovelluskohtaisesti: sovelluksen oikolukuasetuksissa käyttäjä voisi itse kytkeä päälle tilan, jossa yhdys-sanat hyväksytään yhdys-merkillisinä.

Voikko-kirjastossa taitaa jo olla tällainen asetus (accept_extra_hyphens=1), eli sovellusohjelmat voivat tuota hyödyntää.
